shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -1162,10 +1248,8 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
       val columnarBatchRdd = input.mapPartitions(iter => {
         new RowToColumnarIterator(iter, structSchema, RequireSingleBatch, converters)
       })
-      columnarBatchRdd.map(cb => {
-        withResource(cb) { columnarBatch =>
-          val cachedBatch = compressColumnarBatchWithParquet(columnarBatch)
-          cachedBatch
+      columnarBatchRdd.flatMap(cb => {
+        withResource(cb) { cb => compressColumnarBatchWithParquet(cb)
```

Comment:
nit: It is probably cleaner to do it like the following.  At a minimum if the closure can fit on a single line it is ideal to use `()` instead of `{}` and to put it on a single line.

```
withResource(cb)(compressColumnarBatchWithParquet)
```